python-wrapper/src/neo4j_viz/nvl.py

```diff
@@ -54,14 +81,8 @@ def render(
         height: str,
         show_hover_tooltip: bool,
     ) -> HTML:
-        try:
-            nodes_json = json.dumps([node.to_dict() for node in nodes])
-        except TypeError as e:
-            raise self.unsupported_field_type_error(e, "node")
-        try:
-            rels_json = json.dumps([rel.to_dict() for rel in relationships])
-        except TypeError as e:
-            raise self.unsupported_field_type_error(e, "relationship")
+        nodes_json = json.dumps([self._serialize_entity(node) for node in nodes])
```

Comment:
now we call json.dumps twice.

We could also return the json.dumps from serialize function and just chain them.

However, for the given data volume this should also not be a bottle neck

```suggestion
        nodes_json = f"[{[self._serialize_entity(node) for node in nodes].join(',)}]"
```

Reply:
Good idea! Done